src/service.rs

```diff
@@ -91,8 +92,16 @@ impl Service<Request<Body>> for IconService {
         }
         let size = size.parse::<u8>().unwrap();
 
+        let max_misc = params
+            .get("misc")
+            .cloned()
+            .unwrap_or(String::from(DEFAULT_MISC))
+            .parse::<u32>()
+            .unwrap()
```

Comment:
This unwrap will cause a 500 if you're passed a string. It'd be more ideal to catch and return a BAD_REQUEST.

```rust
 let max_misc = match params
            .get("misc")
            .cloned()
            .unwrap_or(String::from(DEFAULT_MISC))
            .parse::<u32>() {
    Ok(misc) => misc.clamp(0, 255) as u8,
    Err(_) => {
        *response.status_mut() = StatusCode::BAD_REQUEST;
        return Box::pin(async { Ok(response) });
    },
};
```

This wasn't possibly above because it compares against fixed values.